lib/rucio/daemons/automatix/automatix.py

```diff
@@ -68,210 +69,216 @@ def choose_element(probabilities: dict, data: str) -> float:
 
-        for rse in rses:
-            start_time = time()
-            tmpdir = tempfile.mkdtemp()
-            logger(logging.INFO, 'Running on RSE %s', rse)
-            dic = choose_element(probabilities, data)
-            metadata = dic['metadata']
-            try:
-                nbfiles = dic['nbfiles']
-            except KeyError:
-                nbfiles = 2
-                logger(logging.WARNING, 'No nbfiles defined in the configuration, will use 2')
-            try:
-                filesize = dic['filesize']
-            except KeyError:
-                filesize = 1000000
-                logger(logging.WARNING, 'No filesize defined in the configuration, will use 1M files')
-            dsn = generate_didname(metadata, None, 'dataset')
-            fnames = []
-            lfns = []
-            physical_fnames = []
-            files = []
-            for _ in range(nbfiles):
-                fname = generate_didname(metadata=metadata, dsn=dsn, did_type='file')
-                lfns.append(fname)
-                logger(logging.INFO, 'Generating file %s in dataset %s', fname, dsn)
-                physical_fname = '%s/%s' % (tmpdir, "".join(fname.split('/')))
-                physical_fnames.append(physical_fname)
-                generate_file(physical_fname, filesize, logger=logger)
-                fnames.append(fname)
-                file_ = {'did_scope': scope, 'did_name': fname, 'dataset_scope': scope, 'dataset_name': dsn, 'rse': rse, 'path': physical_fname}
-                files.append(file_)
-            logger(logging.INFO, 'Upload %s:%s to %s', scope, dsn, rse)
-            upload_client = UploadClient()
-            ret = upload_client.upload(files)
-            if ret == 0:
-                logger(logging.INFO, '%s sucessfully registered' % dsn)
-                client = Client()
-                if set_metadata:
-                    client.set_metadata_bulk(scope=scope, name=dsn, meta=metadata, recursive=False)
-                    monitor.record_counter(name='automatix.addnewdataset.done', delta=1)
-                    monitor.record_counter(name='automatix.addnewfile.done', delta=nbfiles)
-                    monitor.record_timer('automatix.datasetinjection', (time() - start_time) * 1000)
-            else:
-                logger(logging.INFO, 'Error uploading files')
-            for physical_fname in physical_fnames:
-                remove(physical_fname)
-            rmdir(tmpdir)
-        if once is True:
-            logger(logging.INFO, 'Run with once mode. Exiting')
-            break
-        tottime = time() - starttime
-        if status:
-            logger(logging.INFO, 'It took %s seconds to upload one dataset on %s', str(tottime), str(rses))
-            daemon_sleep(start_time=starttime, sleep_time=sleep_time, graceful_stop=GRACEFUL_STOP, logger=logger)
+def run_once(heartbeat_handler: "HeartbeatHandler", inputfile: str, **_kwargs) -> bool:
+
+    _, _, logger = heartbeat_handler.live()
```

Comment:
Why move this code from run() to run_once() in a commit called "Heartbeat handler for automatix daemon". If this is needed for this commit, it must be explained in the commit message. If it's unrelated, it must be in a different commit.

Reply:
Main reason is that the config was resolved once at the start of the daemon, but then never refreshed